app/client/src/widgets/JSONFormWidget/component/Form.tsx

```
@@ -273,6 +273,12 @@ function Form<TValues = any>(
     onFormValidityUpdate(!isFormInValid);
   }, [onFormValidityUpdate, isFormInValid]);
 
+  useEffect(() => {
+    // This effect is to handle an edge-case:
+    // https://github.com/appsmithorg/appsmith/issues/28018
+    trigger();
```

I didn't quite understood the solution here, why are we re-validating the whole form when children changes? Isn't it a problem of when reset button is pressed the validation persists?

---

@ashit-rath I don't think so, the problem is the validation not getting triggered when children changes. When reset button is pressed, then the validation change is triggered.

---

@ashit-rath 
Let me take you with a dry run of whats happening in the original issue #28018
1. User select red and get one input which bound to red. It is empty hence, errors has `{red: required:true}`.
2. user adds value, errors are cleared. and user can submit.
3. Now they click reset, the input is cleared and errorState has this: `{red: required:true}`
4. Now user switches to blue, and gets input bound to blue, which adds `{blue: required:true}`
5. Over all errorState looks liks: `{red: required:true, blue: required:true}`
6. Now even if you type in blue input, the red error never goes away. 

I am open to ideas if this call to `trigger` can be optimized. Looking forward to your ideas.

---

Thanks @rahulbarwal and @jacquesikot for explaining this in detail, I understood the problem. I believe this is a reset problem; if you look at the `RESET_OPTIONS` which gets passed to `onReset` has `keepErrors: true`. Have we checked if this was set to `false` for the case when button is clicked; solves the problem?

---

Thanks @ashit-rath for this input. 
`keepErrors: false` did not work, and I think I know why: 
In step `3` above, when user resets the form, the red input is visible and Here, whether we keep or discard errors does not matter. 
IT is on step `4` that multiple errors enter the formState(where we are not doing the reset).

---

Thanks @ashit-rath, for this feedback, I've incorporated these changes. please review again.